File: Reflow/Provider/SqlServer/SqlDataLink.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using System.Text;

namespace Reflow.Provider.SqlServer
{
    public class SqlDataLink : IDataLink
    {
        SqlConnection _connection = null;
        public bool Initialize(string connectionString)
        {
            _connection = new SqlConnection(connectionString);
            return true;
        }

        public bool IsConnected
        {
            get 
            {
                if (_connection != null)
                {
                    return _connection.State == ConnectionState.Open;
                }
                return false;
            }
        }

        public bool Connect()
        {
            _connection.Open();
            return true;
        }

        public bool Disconnect()
        {
            _connection.Close();
            return false;
        }

        public IDbConnection Connection
        {
            get
            {
                return _connection;
            }
        }

        public ColumnDefinition[] GetSchema(string query)
        {
            List<ColumnDefinition> items = new List<ColumnDefinition>();
            SqlCommand cmd = _connection.CreateCommand();
            cmd.CommandText = query;
            cmd.CommandType = CommandType.Text;
            IDataReader reader = cmd.ExecuteReader(CommandBehavior.SchemaOnly);
            DataTable dt = reader.GetSchemaTable();
            // ColumnName, ColumnSize, NumericPrecision, DataType, AllowDbNull, ProviderType, 
            // IsIdentity, IsAutoIncrement, ProviderSpecificDataType, DataTypeName
            foreach (DataRow row in dt.Rows)
            {
                ColumnDefinition col = new ColumnDefinition();
                col.ColumnName = row["ColumnName"] as string;
                col.Length = (int)row["ColumnSize"];
                //string systemDbTypeName = col[""] as string;
                items.Add(col);
            }
            return items.ToArray();
        }


        public bool CreateTable(string tableName, ColumnDefinition[] columns, bool shouldDropExisting)
        {
            Helper.TableGenerator tgen = new Helper.TableGenerator(new SqlTypeTranslator());
            string script = tgen.GenerateTableScript(tableName, columns);
            SqlCommand cmd = _connection.CreateCommand();
            cmd.CommandType = CommandType.Text;
            if (shouldDropExisting)
            {
                cmd.CommandText = "DROP TABLE " + tableName;
                try
                {
                    cmd.ExecuteNonQuery();
                }
                catch (SqlException ex) { }
            }
            cmd.CommandText = script;
            
            cmd.ExecuteNonQuery();
            return true;
        }
    }
}

File: Java/src/main/java/com/nuarca/etl/provider/sqlserver/SqlDataLink.java
package com.nuarca.etl.provider.sqlserver;

//
// Translated by CS2J (http://www.cs2j.com): 12/6/2017 2:34:21 PM
//




import com.nuarca.etl.ColumnDefinition;
import com.nuarca.etl.helper.TableGenerator;
import com.nuarca.etl.provider.IDataLink;
import com.nuarca.etl.provider.sqlserver.SqlTypeTranslator;

import javax.tools.JavaCompiler;
import java.sql.*;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class SqlDataLink   implements IDataLink
{
    Connection _connection = null;
    public boolean initialize(String connectionString) throws Exception {

        Driver driver = (Driver) Class.forName("com.microsoft.sqlsserver.jdbc.SQLServerDriver").newInstance();
        Connection conn = driver.connect(connectionString, new Properties());
        _connection = conn;
        return true;
    }

    public boolean getIsConnected() throws Exception {
        if (_connection != null)
        {
            return !_connection.isClosed();
        }

        return false;
    }

    public boolean connect() throws Exception {
        return true;
    }

    public boolean disconnect() throws Exception {
        _connection.close();
        return false;
    }

    public Connection getConnection() throws Exception {
        return _connection;
    }

    public ColumnDefinition[] getSchema(String query) throws Exception {
        List<ColumnDefinition> items = new ArrayList<ColumnDefinition>();


        Statement cmd = _connection.prepareCall( query );
        ResultSet reader = cmd.executeQuery(query);
        ResultSetMetaData meta = reader.getMetaData();

        int colCount = meta.getColumnCount();

        for( int i = 0; i < colCount; i++){
            ColumnDefinition col = new ColumnDefinition();
            col.setColumnName(meta.getColumnName(i));
            col.setIsNullable(meta.isNullable(i)==1);
            col.setLength( meta.getPrecision(i));
            items.add(col);

        }
        return ((ColumnDefinition[]) items.toArray());
    }

    public boolean createTable(String tableName, ColumnDefinition[] columns, boolean shouldDropExisting) throws Exception {
        /*
        TableGenerator tgen = new TableGenerator(new SqlTypeTranslator());
        String script = tgen.GenerateTableScript(tableName, columns);
        SqlCommand cmd = _connection.CreateCommand();
        cmd.CommandType = CommandType.Text;
        if (shouldDropExisting)
        {
            cmd.CommandText = "DROP TABLE " + tableName;
            try
            {
                cmd.ExecuteNonQuery();
            }
            catch (SqlException ex)
            {
            }

        }

        cmd.CommandText = script;
        cmd.ExecuteNonQuery();
        */
        return true;
    }

}


